### PR TITLE
Changes for supporting K8S Bound SAT auth in pubsub and store

### PR DESCRIPTION
**Pull request by bilalsellak**
# Description

Change to support K8S Bound SAT authentication in the E4Kmqtt pubsub component and E4K store components
## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ x] Code compiles correctly
* [ x] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pubsub/mqtte4k/metadata.go [hunk 1]
```diff
@@ -71,20 +75,30 @@ func parseMQTTMetaData(md pubsub.Metadata, log logger.Logger) (*metadata, error)
...
-	if val, ok := md.Properties[spiffeBrokerAudience]; ok && val != "" {
-		m.spiffeBrokerAudience = val
+		if val, ok := md.Properties[spiffeBrokerAudience]; ok && val != "" {
+			m.spiffeBrokerAudience = val
+		} else {
+			return &m, fmt.Errorf("%s Invalid or Missing spiffeBrokerAudience", errorMsgPrefix)
+		}
 	} else {
-		return &m, fmt.Errorf("%s Invalid or Missing spiffeBrokerAudience", errorMsgPrefix)
+		m.brokerAuthMethod = "SAT"
```
Comment (jollyaakash) [inline pubsub/mqtte4k/metadata.go:92]: so we are keeping SAT as default right

File: pubsub/mqtte4k/metadata.go [hunk 2]
```diff
@@ -71,20 +75,30 @@ func parseMQTTMetaData(md pubsub.Metadata, log logger.Logger) (*metadata, error)
...
+		if val, ok := md.Properties[satTokenPath]; ok && val != "" {
+			m.satTokenPath = val
+		} else {
+			return &m, fmt.Errorf("%s Invalid or Missing satTokenPath", errorMsgPrefix)
+		}
 	}
 
 	// optional configuration settings
-
+	// make this mandatory
```
Comment (jollyaakash) [inline pubsub/mqtte4k/metadata.go]: why?

File: pubsub/mqtte4k/mqtte4k.go [hunk 3]
```diff
@@ -42,23 +42,37 @@ const (
...
 	return &mqttPubSub{
-		logger:          logger,
+		logger: logger,
 	}
 }
 
+func populateSATPassword(m *mqttPubSub) {
+	token, err := os.ReadFile(m.metadata.satTokenPath)
+	if err != nil {
+		panic(err)
```
Comment (jollyaakash) [inline pubsub/mqtte4k/mqtte4k.go]: maybe add error log here saying, if file is not found. Have you added annotations?

File: pubsub/mqtte4k/mqtte4k.go [hunk 4]
```diff
@@ -42,23 +42,37 @@ const (
...
+	}
+
+	satToken := string(token) // convert token to a String
+
+	m.satToken = satToken
+	m.logger.Debugf("mqtte4k got SAT Token: %s", satToken)
```
Comment (jollyaakash) [inline pubsub/mqtte4k/mqtte4k.go]: remove spitting out satToken

**Comment by jollyaakash** [comment]
Added comments in mqtte4k section, but same logic in e4k

File: pubsub/mqtte4k/metadata.go (hunk 1 above)
Comment (bilalsellak) [reply pubsub/mqtte4k/metadata.go:92]: yes, we should

File: pubsub/mqtte4k/metadata.go (hunk 2 above)
Comment (bilalsellak) [reply pubsub/mqtte4k/metadata.go]: Whoops. This comment was just reminder to me that we always need clientid so the check for it should be mandatory

**Pull request merged**